### PR TITLE
implementation of TACACS+ authorization

### PR DESCRIPTION
**Pull request by evgvain**
I have added the authorization capabilities accordingly to the protocol specification
(https://tools.ietf.org/html/draft-ietf-opsawg-tacacs-06)

There are two new packets implemented: Account REQUEST Packet and Account Reply Packet
There is a new method of TACACSClient authorizate
In addition added a method to close socket, so each new call to authenticate and authorizate will re-create a new socket.

Will kindly appreciate your review and merge to the new version.

Best Regards,
Evgeny

### CODE REVIEW

File: tacacs_plus/__init__.py [hunk 1]
```diff
@@ -220,10 +242,10 @@ def __str__(self):
 
 class TACACSAuthenticationStart(object):
 
-    def __init__(self, username, authen_type, data=six.b('')):
+    def __init__(self, username, authen_type, priv_lvl=TAC_PLUS_PRIV_LVL_MIN, data=six.b('')):
```
Comment (ryanpetrello) [inline tacacs_plus/__init__.py]: Why pass `priv_lvl` to AuthenticationStart if we're not doing anything with it?

File: tacacs_plus/__init__.py [hunk 2]
```diff
@@ -492,14 +661,15 @@ def send(self, body, seq_no=1):
             )
         raise socket.timeout
 
-    def authenticate(self, username, password,
+    def authenticate(self, username, password, priv_lvl=TAC_PLUS_PRIV_LVL_MIN,
```
Comment (ryanpetrello) [inline tacacs_plus/__init__.py:676]: Same comment goes for this parameter; we're passing this parameter in, but doing nothing with in terms of packet building.

**Comment by ryanpetrello** [comment]
Thanks for contributing this :)!

I only have a few minor comments on the pull request.

File: tacacs_plus/__init__.py (hunk 1 above)
Comment (evgvain) [reply tacacs_plus/__init__.py]: Although, this parameter is related to authorization, it is required as a part of authentication start packet. I assumed that some server implementation might address it even on authentication. So, my suggestion is to allow passing it as a parameter.

File: tacacs_plus/__init__.py (hunk 2 above)
Comment (evgvain) [reply tacacs_plus/__init__.py:676]: My idea was to support the case, when this parameter is passed to authenticate method and used in  authentication start packet.

File: tacacs_plus/__init__.py (hunk 1 above)
Comment (ryanpetrello) [reply tacacs_plus/__init__.py]: Ah, I see - I misread the diff; you're changing the previously hard-coded value from `TAC_PLUS_PRIV_LVL_MIN`

**Comment by ryanpetrello** [comment]
@evgvain the tests were failing w/ your PR;  https://github.com/SecurityDam/tacacs_plus/pull/1 should fix them up

if you'll merge that change and include it in this PR, I'll merge this support in and work on a PyPI release

**Comment by ryanpetrello** [comment]
@evgvain thanks again for contributing!

https://pypi.python.org/pypi/tacacs_plus/0.2

**Comment by evgvain** [comment]
Thank you!

**Pull request merged**